include/swift/AST/TypeCheckRequests.h

```diff
@@ -2426,9 +2426,33 @@ class HasImplementationOnlyImportsRequest
   bool isCached() const { return true; }
 };
 
+struct TypeResolutionDescriptor final {
```

Comment:
I'd still like this descriptor broken back down. We only really use them when it's annoying/impossible to thread types through the evaluator.

Reply:
Okay. It would be great if requests could provide a custom print implementation in terms of simple_display, though.